src/java.base/share/classes/jdk/internal/classfile/TypeKind.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.classfile;
+
+/**
+ * Describes the types that can be part of a field or method descriptor.
+ */
+public enum TypeKind {
+    /** the primitive type byte */
+    ByteType("byte", "B", 8),
+    /** the primitive type short */
+    ShortType("short", "S", 9),
+    /** the primitive type int */
+    IntType("int", "I", 10),
+    /** the primitive type float */
+    FloatType("float", "F", 6),
+    /** the primitive type long */
+    LongType("long", "J", 11),
+    /** the primitive type double */
+    DoubleType("double", "D", 7),
+    /** a reference type */
+    ReferenceType("reference type", "A", -1),
```

Comment:
```suggestion
    ReferenceType("reference type", "L", -1),
```
?
Otherwise `TypeKind.fromDescriptor(TypeKind.ReferenceType.descriptor())` will fail.